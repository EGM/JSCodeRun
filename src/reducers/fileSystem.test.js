import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import reducer from './fileSystem';

describe('reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, { type: 'not an action' })).toMatchSnapshot();
  });

  it('should return state after file save success', () => {
    expect(reducer(undefined, { type: 'FILE_SAVE_SUCCESS' })).toMatchSnapshot();
  });

  it('should return state after file save failure', () => {
    expect(reducer(undefined, { type: 'FILE_SAVE_FAILURE', err: 'save error' })).toMatchSnapshot();
  });

  it('should return state after file delete failure', () => {
    expect(reducer(undefined, { type: 'FILE_DELETE_FAILURE', err: 'delete error' })).toMatchSnapshot();
  });

  it('should return state after file open success', () => {
    expect(reducer(undefined, { type: 'FILE_OPEN_SUCCESS' })).toMatchSnapshot();
  });

  it('should return state after file open failure', () => {
    expect(reducer(undefined, { type: 'FILE_OPEN_FAILURE', err: 'open error' })).toMatchSnapshot();
  });

  it('should return state after edit code', () => {
    expect(reducer(undefined, { type: 'EDIT_CODE' })).toMatchSnapshot();
  });
});
