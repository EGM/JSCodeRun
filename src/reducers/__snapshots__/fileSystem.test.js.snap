// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer should return state after edit code 1`] = `
Object {
  "currentFile": "",
  "error": null,
  "isEdited": true,
}
`;

exports[`reducer should return state after file delete failure 1`] = `
Object {
  "currentFile": "",
  "error": null,
}
`;

exports[`reducer should return state after file open failure 1`] = `
Object {
  "currentFile": "",
  "error": "open error",
}
`;

exports[`reducer should return state after file open success 1`] = `
Object {
  "currentFile": undefined,
  "error": null,
  "isEdited": false,
}
`;

exports[`reducer should return state after file save failure 1`] = `
Object {
  "currentFile": "",
  "error": "save error",
}
`;

exports[`reducer should return state after file save success 1`] = `
Object {
  "currentFile": undefined,
  "error": null,
  "isEdited": false,
}
`;

exports[`reducer should return the initial state 1`] = `
Object {
  "currentFile": "",
  "error": null,
}
`;
